xds/src/test/java/io/grpc/xds/XdsNameResolverTest.java

```diff
@@ -260,6 +260,7 @@ public void resolve_foundResource() {
     assertThat(result.getAddresses()).isEmpty();
     Map<String, ?> serviceConfig =
         result.getAttributes().get(GrpcAttributes.NAME_RESOLVER_SERVICE_CONFIG);
+    verify(serviceConfigParser).parseServiceConfig(eq(serviceConfig));
```

Comment:
nit: I think the goal of the test should be verifying `result.getServiceConfig()` is updated.

```java
Object fakeParsedConfig = new Object();
when(serviceConfigParser.parseServiceConfig(anyMap)).return(ConfigOrError.fromConfig(fakeParsedConfig));
...
verify(serviceConfigParser).parseServiceConfig(eq(serviceConfig));
assertThat(result.getServiceConfig().getConfig()).isEqualTo(fakeParsedConfig);
```

Reply:
Changed a bit. This should cover the invocation of `ResolutionResult#setServiceConfig(...)`.